Fluid/Filters/MiscFilters.cs

```diff
@@ -160,13 +165,72 @@ public static FluidValue Date(FluidValue input, FilterArguments arguments, Templ
                     {
                         switch (c)
                         {
-                            case 'a': result.Append(context.CultureInfo.DateTimeFormat.AbbreviatedDayNames[(int)value.DayOfWeek]); break;
-                            case 'A': result.Append(context.CultureInfo.DateTimeFormat.DayNames[(int)value.DayOfWeek]); break;
-                            case 'b': result.Append(context.CultureInfo.DateTimeFormat.AbbreviatedMonthNames[value.Month - 1]); break;
-                            case 'B': result.Append(context.CultureInfo.DateTimeFormat.MonthNames[value.Month - 1]); break;
-                            case 'c': result.Append(value.ToString("F", context.CultureInfo)); break;
+                            case '^': upperCaseFlag = true; continue;
+                            case '-': removeLeadingZerosFlag = true; continue;
+                            case '_': useSpaceForPaddingFlag = true; continue;
+                            case ':': useColonsForZeeDirectiveFlag = true; continue;
+                            case 'a':
+                            {
+                                result.Append(
+                                    upperCaseFlag
```

Comment:
Could you rewrite your changes by reusing the computational code only once, like this:

```
var dayName = context.CultureInfo.DateTimeFormat.AbbreviatedDayNames[(int) value.DayOfWeek];
result.Append(upperCaseFlag ? dayName.ToUpper() : dayName); 
```

Or use an `if` statement if there are more than two cases.

Thanks